components/script/dom/webgl2renderingcontext.rs

```diff
@@ -1420,6 +1435,153 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
+                            self.base.webgl_error(InvalidOperation);
+                            return;
+                        }
+                        current_tf.begin(&self.base, primitiveMode);
+                    },
+                    // GL_INVALID_OPERATION is generated by
+                    // glBeginTransformFeedback if any binding point
+                    // used in transform feedback mode does not have a
+                    // buffer object bound. In interleaved mode,
+                    // only the first buffer object binding point is ever written to.
+                    //
+                    // GL_INVALID_OPERATION is generated by
+                    // glBeginTransformFeedback if no binding
+                    // points would be used, either because no
+                    // program object is active of because the active
+                    // program object has specified no varying variables to record.
+                    None => {
+                        self.base.webgl_error(InvalidOperation);
+                        return;
+                    }
+                }
+            },
+            _ => self.base.webgl_error(InvalidEnum),
+        };
+    }
+
+    fn EndTransformFeedback(&self) {
+        if let Some(current_tf) = self.current_transform_feedback.get() {
+            if !current_tf.is_active() {
+                self.base.webgl_error(InvalidOperation);
+                return;
+            }
+            current_tf.end(&self.base);
+        }
+    }
+
+    fn ResumeTransformFeedback(&self) {
+        if let Some(current_tf) = self.current_transform_feedback.get() {
+            if !current_tf.is_active() || !current_tf.is_paused() {
+                self.base.webgl_error(InvalidOperation);
+                return;
+            }
+            current_tf.resume(&self.base);
+        }
+    }
+
+    fn PauseTransformFeedback(&self) {
+        if let Some(current_tf) = self.current_transform_feedback.get() {
+            if !current_tf.is_active() || current_tf.is_paused() {
+                self.base.webgl_error(InvalidOperation);
+                return;
+            }
+            current_tf.pause(&self.base);
+        }
+    }
+
+    fn TransformFeedbackVaryings(&self, program: &WebGLProgram, varyings: Vec<DOMString>, bufferMode: u32) {
+        handle_potential_webgl_error!(self.base, program.validate(), return);
+        let strs = varyings.iter().map(|name| {
+            name.get_str()
+        }).collect::<Vec<String>>();
+        match bufferMode {
+            constants::INTERLEAVED_ATTRIBS => self.base.send_command(WebGLCommand::TransformFeedbackVaryings(program.id(), strs, bufferMode)),
+            constants::SEPARATE_ATTRIBS => {
+                let (sender, receiver) = webgl_channel().unwrap();
+                match handle_potential_webgl_error!(self.base, Parameter::from_u32(constants::MAX_TRANSFORM_FEEDBACK_SEPARATE_ATTRIBS), return) {
```

Comment:
We should fetch this as part of the GL limits when the context is created so it's always available.